src/pages/Store.tsx

```diff
@@ -15,47 +15,113 @@ import BarcodeIcon from "../components/icons/BarcodeIcon";
 import { FindLocationByBarcodeQuery, Maybe } from "../types/sdk";
 import LoadingSpinner from "../components/icons/LoadingSpinner";
 import { isLocationSearch, LocationSearchParams } from "../types/stan";
-import { history, StanCoreContext } from "../lib/sdk";
+import { StanCoreContext } from "../lib/sdk";
 import { ClientError } from "graphql-request";
-
+import LabwareAwaitingStorage from "./location/LabwareAwaitingStorage";
+import * as H from "history";
+import { history } from "../lib/sdk";
+import PromptOnLeave from "../components/notifications/PromptOnLeave";
 /**
  * RouteComponentProps from react-router allows the props to be passed in
  */
 interface StoreProps extends RouteComponentProps {}
 
+export type LabwareAwaitingStorageInfo = {
+  barcode: string;
+  labwareType: string;
+};
+
+/**
+ *
+ * @param location - location to navigate
+ * @param action - action performed
```

Comment:
A comment in here explaining what PUSH and POP indicate would be useful.

Reply:
Done